### PR TITLE
go/staking: include LastBlockFees in genesis

### PR DESCRIPTION
**Pull request by ptrus**
Also adds a dump-restore version of the gas-fees test.

### CODE REVIEW

**Comment by Yawning** [comment]
Will this break dump/restore?

**Comment by ptrus** [comment]
yeah in the first begin block the `GetLastCommitInfo` will have no votes, so we probably need to also carry over that as well? (or handle differently the initial fee disbursement)

File: go/oasis-test-runner/scenario/e2e/gas_fees_staking.go [hunk 1]
```diff
@@ -106,8 +148,13 @@ func (sc *gasFeesImpl) Run(childEnv *env.Env) error {
 		return err
 	}
 
+	// Include fees from genesis.
+	totalFees, err := sc.getInitialCommonPoolBalance(ctx)
```
Comment (pro-wh) [inline go/oasis-test-runner/scenario/e2e/gas_fees_staking.go]: does this require that the entire common pool come from fees or something?

File: .changelog/2777.feature.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+go/staking: include LastBlockFees in genesis
```
Comment (pro-wh) [inline .changelog/2777.feature.md]: oh upgrade this to breaking

File: go/oasis-test-runner/scenario/e2e/gas_fees_staking.go (hunk 1 above)
Comment (ptrus) [reply go/oasis-test-runner/scenario/e2e/gas_fees_staking.go]: no doesn't matter, in any case common pool will get distributed to nodes. Updated the comment to not explicitly mention fees and also updated the staking-genesis doc used to have both non-empty initial `last_block_fees` and `common_pool`

File: .changelog/2777.feature.md (hunk 2 above)
Comment (ptrus) [reply .changelog/2777.feature.md]: hm so not sure if this is the case, under the "breaking" label in github we have a note: "Breaking change requiring state wipe", i don't think that is the case here, also a genesis without last_block_fees does still work. However the change does break forward compatibility as genesis files with last_block_fees field would not work with old versions :thinking:

Comment (pro-wh) [reply .changelog/2777.feature.md]: oh ok

**Pull request merged**